lib/gollum/app.rb

```diff
@@ -462,7 +462,12 @@ class App < Sinatra::Base
       post '/compare/*' do
         @file     = clean_url(encodeURIComponent(params[:splat].first))
         @versions = params[:versions] || []
-        if @versions.size < 2
+        if @versions.size == 1
+          wikip  = wiki_page(params[:splat].first)
+          commit = wikip.wiki.repo.rev_parse(@versions.first)
+          @versions.push commit.parent_ids.first
+        end
+        if @versions.empty?
```

Comment:
This could be an `elsif`

Reply:
I don't think so. if `@version.size == 1`, the parent of `@version.first` is pushed to `@version` and, we want to enter the next `else` block. This will not happen with `elsif @versions.empty?` here.